src/test/java/org/apache/commons/codec/digest/MurmurHash3Test.java

```diff
@@ -224,4 +224,68 @@ public void testSingleLongRandom() {
+
+        byte[] bytes = "Now is the time for all good men to come to the aid of their country".getBytes("UTF-8");
+	      int hash=0;
+	      for (int i=0; i<bytes.length; i++) {
+	        hash = hash*31+(bytes[i]&0xff);
+	        bytes[i] = (byte)hash;
+	      }
+
+	      // test different offsets.
+	      for (int offset = 0; offset<20; offset++) {
+	        // put the original bytes at the offset so the same hash will be generated
+	        byte[] arr = new byte[bytes.length + offset];
+	        System.arraycopy(bytes, 0, arr, offset, bytes.length);
+	        int seed = 1;
+	        long[] result;
+	        for (int len=0; len<bytes.length; len++) {
+	          seed *= 0x9e3779b1;
+	          result = MurmurHash3.hash128(arr, offset, len, seed);
+	          assertEquals(answers128[len * 2], result[0]);
+	          assertEquals(answers128[len*2+1], result[1]);
+	        }
+	      }
+	    }
+
+    /**
+     * Execute the Murmur3_x86_32 tests from Yonic Seeley's implementation tests.
+     * Part of CODEC-264 testing
+     * @see https://github.com/yonik/java_util/blob/master/test/util/hash/TestMurmurHash3.java
+     * @throws Exception
+     */
+	@Test
+    public void testHash32SignExtension() throws Exception {
+        int[] answers32 = new int[] {0x11fd02eb,0x8dd65a73,0x29b074ba,0xcbcd43ce,0xb6463881,0xf6228557,0x3d55c634,0xa1bb9072,0x448402c6,0xb12bf3d4,0x18a71ccb,0x6ae5f185,0x9a482256,0xc686d7f2,0x8e8984d8,0x68a2491d,0xcc29b0e6,0x3e9130bd,0xc90defb3,0xf81c5978,0x15ff7f63,0x4ec16a7a,0xa08aa899,0x7317ffee,0x93752d34,0x400f8781,0x2358838c,0x6ecb8998,0x45a5c102,0x46ed68fd,0xfecb51c0,0x7a68c7db,0x9e334eab,0x21ea13b6,0xf184e92c,0xc016220d,0x7f6c9713,0x1e909123,0xb51a21b7,0x94c58881,0xe4e91bf0,0xde80a366,0xfd84005a,0x3361d373,0xe7d528cc,0x487275a7,0xf2290ee5,0x869992a8,0x63cdd341,0x8e94b334,0x1fc7bf11,0x5228b0,0xb4292b62,0x36ed3770,0xfe914519,0x7d9d1830,0xe1acfb60,0xc8b4d4b7,0xf1ec49ba,0xedbb8cc1,0xdc5b3ab1,0x7c7778ae,0x52bf68d,0xe0bb4148,0xfea36521,0xa0696ca5,0xf28df752,0xd82dccb6};
+
+        byte[] bytes = "Now is the time for all good men to come to the aid of their country".getBytes("UTF-8");
+          int hash=0;
+          for (int i=0; i<bytes.length; i++) {
+            hash = hash*31+(bytes[i]&0xff);
+            bytes[i] = (byte)hash;
+          }
+
+          // test different offsets.
+          for (int offset = 0; offset<20; offset++) {
+            // put the original bytes at the offset so the same hash will be generated
+            byte[] arr = new byte[bytes.length + offset];
+            System.arraycopy(bytes, 0, arr, offset, bytes.length);
+            int seed = 1;
+            for (int len=0; len<bytes.length; len++) {
+              seed *= 0x9e3779b1;
+              int h = MurmurHash3.hash32(arr, offset, len, seed);
+              assertEquals(answers32[len], h);
+            }
+          }
+        }
+
```

Comment:
The formatting of these blocks is off. The original code uses tabs. Can you stick to the tab style or use 4 character space indents instead.

MurmurHash2/3 and their tests are the only files in the repo that use tabs. All others use 4 character indentation. This can be fixed at another time.

Reply:
In general, we are a no tabs project, see "Spaces No Tabs" here https://commons.apache.org/patches.html. 

It was my mistake to bring in the original PR with tabs in the first place, dang. We can fix tabs separately indeed, whatever makes the process best for review...

Reply:
That was my fault I should have scrubbed the tabs when I brought this file over.